### PR TITLE
MCR-5296 Update API Authorization for OAuth Tokens

### PR DESCRIPTION
**Pull request by shamoons**
## Summary

This pull request introduces support for OAuth tokens in the authentication system, alongside improvements to the JWT library and its integration across the codebase. The changes include adding methods for creating and validating OAuth tokens, updating existing logic to handle both OAuth and standard tokens, and enhancing test coverage for the new functionality.

### OAuth Token Support

- Added `createOAuthJWT` and `validateOAuthToken` methods to the `JWTLib` interface for generating and validating OAuth tokens. These methods handle tokens with `client_id` and `grant_type` claims. (`services/app-api/src/jwt/jwt.ts`)
- Updated the `third_party_API_authorizer` handler to first attempt validation as an OAuth token and fall back to standard token validation if OAuth validation fails. (`services/app-api/src/handlers/third_party_API_authorizer.ts`)

### JWT Library Enhancements

- Refactored the `newJWTLib` function to include OAuth-specific methods and updated the `JWTLib` interface accordingly.
- Introduced the `OAuthTokenPayload` interface to define the structure of OAuth token claims.

### Integration with OAuth2 Server

- Replaced the manual JWT generation logic in the `CustomOAuth2Server` class with the new `createOAuthJWT` method from the `JWTLib`. This ensures consistency in token creation and leverages the updated library.

### Test Coverage

- Added new test cases for OAuth token creation and validation, including edge cases for invalid tokens and mismatched types. These tests ensure the robustness of the new functionality.

#### Related issues

- [MCR-5296](https://jiraent.cms.gov/browse/MCR-5296)

#### Screenshots

_N/A_

#### Test cases covered

- `jwt.test.ts`: tests for `createOAuthJWT` and `validateOAuthToken`
  - Valid token creation and decoding
  - Rejection of invalid or malformed tokens
  - Fallback behavior in `third_party_API_authorizer`

## QA guidance

1. Run all tests to confirm functionality (`jwt.test.ts`).
2. Invoke the `third_party_API_authorizer` handler with both OAuth and standard tokens to confirm fallback logic.
3. Confirm that `CustomOAuth2Server` issues tokens using the updated `JWTLib`.
4. Validate that the new `OAuthTokenPayload` claims are correctly embedded and decoded.

### CODE REVIEW

File: services/app-api/src/handlers/third_party_API_authorizer.ts [hunk 1]
```diff
@@ -25,12 +25,32 @@ const jwtLib = newJWTLib({
     expirationDurationS: 90 * 24 * 60 * 60, // 90 days
 })
 
-const main: APIGatewayTokenAuthorizerHandler = async (
+const oauthJwtLib = newJWTLib({
+    issuer: 'mcreview-oauth',
+    signingKey: Buffer.from(jwtSecret, 'hex'),
+    expirationDurationS: 3600, // 1 hour
```
Comment (pearl-truss) [inline services/app-api/src/handlers/third_party_API_authorizer.ts:31]: @shamoons apologies if this was already discussed. This will change the expiration duration from 90 days to 1 hour. Is the ARMs team aware of this change?

Comment (shamoons) [reply services/app-api/src/handlers/third_party_API_authorizer.ts]: Done
